packages/entrykit/src/onboarding/GasBalance.tsx

```diff
@@ -10,20 +11,40 @@ import { RelayChains, StepContentProps } from "./common";
 import { TruncatedHex } from "../ui/TruncatedHex";
 import { useShowMutationError } from "../errors/useShowMutationError";
 import { useShowQueryError } from "../errors/useShowQueryError";
+import { usePrevious } from "../errors/usePrevious";
+import { CopyIcon } from "../icons/CopyIcon";
+import { CheckIcon } from "../icons/CheckIcon";
+import { useQueryClient } from "@tanstack/react-query";
 
 export type Props = StepContentProps & {
   sessionAddress: Hex;
 };
 
 export function GasBalance({ isActive, isExpanded, sessionAddress }: Props) {
+  const queryClient = useQueryClient();
   const { chain } = useEntryKitConfig();
+  const [copied, setCopied] = useState(false);
 
   const balance = useShowQueryError(useBalance({ chainId: chain.id, address: sessionAddress }));
+  const prevBalance = usePrevious(balance.data);
   useWatchBlockNumber({ onBlockNumber: () => balance.refetch() });
 
   const setBalance = useShowMutationError(useSetBalance());
   const relayChain = (relayChains as RelayChains)[chain.id];
 
+  const handleCopy = () => {
+    navigator.clipboard.writeText(sessionAddress);
+
+    setCopied(true);
+    setTimeout(() => setCopied(false), 2000);
+  };
+
+  useEffect(() => {
+    if (balance.data != null && prevBalance?.value === 0n && balance.data.value > 0n) {
+      queryClient.invalidateQueries({ queryKey: ["getPrerequisites"] });
+    }
+  }, [balance.data, prevBalance, setBalance, sessionAddress, queryClient]);
```

Comment:
is this change related?

Reply:
Somehow this change wasn't visible locally, must have messed something up while branching and missed it. Removed, this was unintentional.